client/src/domain/student/api/bookmarks.ts

```diff
@@ -0,0 +1,101 @@
+import { createApi } from "@/shared/api/core/createApi";
+import type { components } from "@/shared/types/openapi";
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+
+/* Endpoint 정의 */
+const bookmarksApi = createApi("/bookmarks");
+
+/* Bookmark 관련 타입 정의 */
+export type LectureBookmarkRequest =
+  components["schemas"]["LectureBookmarkRequest"];
+export type LectureBookmarkResponse =
+  components["schemas"]["LectureBookmarkResponse"];
+export type PageLectureBookmarkResponse =
+  components["schemas"]["PageLectureBookmarkResponse"];
+export type Pageable = components["schemas"]["Pageable"];
+
+/* Lecture Bookmark */
+// 내 북마크 목록 조회
+const getMyBookmarks = (pageable: Pageable) =>
+  bookmarksApi.get<PageLectureBookmarkResponse>("", { pageable });
+
+// 북마크 추가
+const addBookmark = (data: LectureBookmarkRequest) =>
+  bookmarksApi.post<LectureBookmarkResponse, LectureBookmarkRequest>("", data);
+
+// 북마크 여부 확인
+const checkBookmark = (lectureId: number) =>
+  bookmarksApi.get<{ [key: string]: boolean }>(`/check/${lectureId}`);
+
+// 북마크 삭제
+const removeBookmarkByBookmarkId = (bookmarkId: number) =>
+  bookmarksApi.del<void>(`/${bookmarkId}`);
+
+const removeBookmarkByLectureId = (lectureId: number) =>
+  bookmarksApi.del<void>(`/lecture/${lectureId}`);
+
+// ========================
+//   * bookmarks API 훅 *
+// ========================
+
+// 내 북마크 목록 조회
+export const useMyBookmarks = (pageable: Pageable) => {
```

Comment:
```suggestion
export const useGetMyBookmarks = (pageable: Pageable) => {
```
p3
api 함수랑 명칭을 통일해도 좋을 것 같아요